.github/workflows/libclang-python-tests.yml

```diff
@@ -30,10 +30,15 @@ jobs:
   check-clang-python:
     # Build libclang and then run the libclang Python binding's unit tests.
     name: Build and run Python unit tests
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.11"]
```

Comment:
What's the reasoning for using 3.7 as a baseline rather than the current minimum version for LLVM, 3.6?

Reply:
Yeah, if we need to lock in an old version of Python, version 3.7 seems like the best bang for your buck for python type annotation features. Plus that still covers Debian old old stable ([Debian buster](https://wiki.debian.org/Python#Supported_Python_Versions)). Version 3.6 restricts the annotation in a way that would go against most python type annotations best practices.

Honestly, I think 3.8 would be better, since that gets you the python `Literal` type, but it doesn't seem like [RFC Upgrading LLVM’s minimum required Python version](https://discourse.llvm.org/t/rfc-upgrading-llvms-minimum-required-python-version/67571) was going anywhere. It is also worth noting that there are instances of new features being used in other python file in LLVM. I just think no one is testing against 3.6?

I explained the rationale and feature breakdown in the first two comments of https://github.com/llvm/llvm-project/issues/76664.

Reply:
We can't just ship the library with annotations that require Python 3.7+ when the listed minimum version is Python 3.6. It's probably going to be difficult to get the default version upgraded given how diverse the supported-platform base is for LLVM, but it might be more feasible to carve out an exception specifically for the libclang Python API.

Just because newer features are used in tip of tree doesn't mean they should be. I'd also assume these aren't in the critical path anywhere as I'm reasonably certain a lot of buildbots are still running Buster with Python 3.7.

Reply:
Hmmm. We are in a bit of a catch-22, since we can't even test Python 3.6 with the `ubuntu-latest` runner.
https://github.com/linux4life798/llvm-project/actions/runs/7435926153

Reply:
Although I would be happy with an exception for the libclang binding to use a min of 3.7, I think the general benefit of using 3.7's level of type annotations still holds for all LLVM python. Maybe we should revisit that RFC about bumping python minimum, but propose 3.7, instead?

Reply:
It looks like Python 3.6 is only supported on Github actions on ubuntu 20.04. Otherwise you basically have to compile it from source. I don't think it's too difficult to setup the job matrix to work with combinations of OSes and python versions?

I think it would be easier to get traction on an RFC by limiting it in scope to the libclang bindings as that should cause significantly less issues than all of LLVM. I definitely wouldn't mind bumping the minimum version across the whole project though...

Reply:
Hmm. One downside to testing on a different OS than other jobs is that the cache might always be cold, since no other CI jobs would keep it alive. So, this python binding test might take at least an hour, instead of the 13min we could see now.

Reply:
Ah. Didn't think about that. Keeping the version at 3.7 in the CI should be fine. But there should be a documented version increase for the libclang python bindings before using any features that a version higher than 3.6.

Reply:
Can't we use pyenv or something like that to test python3.6? If we specify 3.6 we should try to test with that IMHO. If it's not available to be used anymore I think we should RFC a update.

Reply:
I agree, we should be testing the minimum compatibility, but I think it is a bit counter productive to add novel ways of testing Python 3.6.  Even Python 3.7 is considered end of life, but it is still easy available in the GitHub runners and Debian.

https://discourse.llvm.org/t/rfc-upgrading-llvms-minimum-required-python-version/67571

Reply:
* I added a note in the libclang python top level README about needing Python version 3.7.
https://github.com/llvm/llvm-project/pull/77219/commits/58bb2caa0fc4f3bc6b2cd187e524b9cc791355dd
* I also tried to spark a conversation in the RFC to bump python version across LLVM.
https://discourse.llvm.org/t/type-annotations-for-libclang-python-bindings/70644/14

Regardless, testing this library with an older version goes a long way towards ensuring backward compatibility.

Reply:
Seems like we have come to a conclusion on these topics through other comments, so I'm resolving this conversation. Feel free to reopen if I missed something.